src/jit/gentree.cpp

```diff
@@ -18017,9 +18017,18 @@ bool GenTree::isRMWHWIntrinsic(Compiler* comp)
         HWIntrinsicFlag flags = Compiler::flagsOfHWIntrinsic(AsHWIntrinsic()->gtHWIntrinsicId);
         return ((flags & HW_Flag_NoRMWSemantics) == 0);
     }
-#endif // _TARGET_XARCH_
 
+    switch (AsHWIntrinsic()->gtHWIntrinsicId)
```

Comment:
FYI. @CarolEidt, this was added to cover https://github.com/dotnet/coreclr/issues/16848

Reply:
`!isSIMD()` is not enough?

Reply:
That would probably work, but it also ends up being over-conservative. For example, it would needlessly cover LZCNT and POPCNT which are always `W, R` (CRC32 is `RW, R`)

Reply:
I'm not crazy about fixing this with a switch statement - I'd like for this to be handled via the table, for consistency, and so that it will be explicitly seen as something to consider setting when new instructions are added.

Reply:
@CarolEidt, would that be a new flag `RMWSemanticsUnderVex` (or some better name)?

Reply:
Right - perhaps this is something to leave for future, but my preference would be to have something like `RMWSemanticsSseOnly` and `RMWSemanticsAlways` or something like that.

Reply:
But I'm OK with deferring if you open an issue.

Reply:
I have logged https://github.com/dotnet/coreclr/issues/16852 to track the issue.

I have also started work on this issue, but will submit it in a follow up PR.

Reply:
Sounds good. Thanks!

Reply:
Sounds good. Thanks!